File: Lua_5.4/1140 - Flores Florescem da França.lua
--- author: Josélio de S. C. Júnior <joseliojrx25@gmail.com>
--- copyright: Josélio de S. C. Júnior 2021

local a = {}
while true do
  local r = io.read()
  if r == '*' then break end
  local t = {}
  for i in r:gmatch '[^ ]+' do
    table.insert(t, string.lower(i:match '%a'))
  end
  local res = 0
  for i = 2, #t, 1 do
    if t[1] ~= t[i] then res = res + 1 end
  end
  res = res == 0 and 'Y' or 'N'
  table.insert(a, res)
end
print(table.concat(a, '\n'))

File: C#_mono_5.10.1.20/1140 - Flores Florescem da França.cs
/// author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
/// copyright Josélio Júnior (Lunatic Fox) 2022

using System;
using System.Linq;

class URI {
  static void Main(string[] args) { 
    while (true) {
      string r = Console.ReadLine() ?? "";
  
      if (r == "*") break;
  
      string[] t = r.Split(' ')
        .Select(e => e[0].ToString().ToLower())
        .ToArray();
  
      if (t.All(e => e == t[0])) {
        Console.WriteLine("Y");
      } else Console.WriteLine("N");
    }
  }
}
